src/math/trig.c

```diff
@@ -36,6 +36,72 @@ const int32_t cord_sincos_piovertwo_q29fl  = Q_CONVERT_FLOAT(_M_PI / 2, 29);
 /**
  * \CORDIC-based approximation of sine and cosine
  */
+#ifdef CORDIC_HIFI3
+void cordic_sin_cos(int32_t th_rad_fxp, cordic_cfg type, int32_t *sign, int32_t *b_yn, int32_t *xn,
+		    int32_t *th_cdc_fxp)
+
+{
+	ae_f32x2 xy_tmp;
+	ae_f32x2 yn_xn;
+	int32_t b_idx;
+
+	int32_t a_idx = CORDIC_31B_TABLE_SIZE;
+	*sign = 1;
+	/* Addition or subtraction by a multiple of pi/2 is done in the data type
+	 * of the input. When the fraction length is 29, then the quantization error
+	 * introduced by the addition or subtraction of pi/2 is done with 29 bits of
+	 * precision.Input range of cordicsin must be in the range [-2*pi, 2*pi),
+	 * a signed type with fractionLength = wordLength-4 will fit this range
+	 * without overflow.Increase of fractionLength makes the addition or
+	 * subtraction of a multiple of pi/2 more precise
+	 */
+	if (th_rad_fxp > cord_sincos_piovertwo_q28fl) {
+		if ((th_rad_fxp - cord_sincos_piovertwo_q29fl) <= cord_sincos_piovertwo_q28fl) {
+			th_rad_fxp -= cord_sincos_piovertwo_q29fl;
+			*sign  = -1;
+		} else {
+			th_rad_fxp -= cordic_sine_cos_piovertwo_q30fl;
+		}
+	} else if (th_rad_fxp < -cord_sincos_piovertwo_q28fl) {
+		if ((th_rad_fxp + cord_sincos_piovertwo_q29fl) >= -cord_sincos_piovertwo_q28fl) {
+			th_rad_fxp += cord_sincos_piovertwo_q29fl;
+			*sign  = -1;
+		} else {
+			th_rad_fxp += cordic_sine_cos_piovertwo_q30fl;
+		}
+	}
+
+	th_rad_fxp = AE_SLAI32(th_rad_fxp, 2);
+	yn_xn = AE_SEL32_LL(0, cordic_sine_cos_lut_q29fl);
```

Comment:
Can you please confirm `Intrinsic AE_SEL32LL does not exist in the current configuration.` ?

Reply:
I check the implementation using hifi_bd5

Reply:
You should use the core configurations for Intel APL, CNL, TGL, etc. platforms. In CI build of this PR passed for all platforms.